presto-hive/src/main/java/com/facebook/presto/twitter/hive/thrift/HiveThriftFieldIdResolver.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.twitter.hive.thrift;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import io.airlift.log.Logger;
+
+import java.util.Properties;
+
+public class HiveThriftFieldIdResolver
+        implements ThriftFieldIdResolver
+{
+    private static final Logger log = Logger.get(HiveThriftFieldIdResolver.class);
+    private static final ObjectMapper objectMapper = new ObjectMapper();
+    public static final String THRIFT_FIELD_ID_JSON = "thrift.field.id.json";
+    private final JsonNode root;
+
+    public HiveThriftFieldIdResolver()
+    {
+        this.root = null;
+    }
+
+    public HiveThriftFieldIdResolver(JsonNode root)
+    {
+        this.root = root;
+    }
+
+    public ThriftFieldIdResolver initialize(Properties schema)
+    {
+        String jsonData = schema.getProperty(THRIFT_FIELD_ID_JSON);
+        try {
+            return new HiveThriftFieldIdResolver(objectMapper.readTree(jsonData));
+        }
+        catch (Exception e) {
```

Comment:
This pattern (of catching Exception) is really bad, because it encapsulates anything that could possibly go wrong and assigns a single solution to it. If you want to handle the case where the json is not presented, handle it with a null check. Same for the case where it might be corrupted, handle it by catching the expected exception that would be thrown in that case (JsonProcessingException?) and logging it.

What if you had corrupt json in the system due to a bug in dalsync? The current approach would mask that forever as an expected condition (you have to assume we're not running in production with debug logging enabled, and even if we were the scenario I describe is worthy of error logging).